src/Control/Monad/Apiary/Action/Internal.hs

```diff
@@ -455,50 +469,97 @@ params = QuasiQuoter
     , quoteDec  = error "params QQ is defined only exp."
     }
 
+-- | only get parameters in query string.
+getQueryParams :: Monad m => ActionT exts prms m HTTP.Query
+getQueryParams = Wai.queryString <$> getRequest
+
 getDocuments :: Monad m => ActionT exts prms m Documents
 getDocuments = liftM actionDocuments getEnv
 
--- | parse request body and return it. since 1.2.2.
-getReqBody :: MonadIO m => ActionT exts prms m RequestBody
+-- | parse request body into 'ActionReqBody' and return it. since 1.2.2.
+getReqBody :: MonadIO m => ActionT exts prms m ActionReqBody
 getReqBody = ActionT $ \_ e s c -> case actionReqBody s of
     Just  b -> c b s
     Nothing -> do
         let req  = actionRequest e
-            body = Wai.requestBody req
-        b <- liftIO $ case P.getRequestBodyType req of
-            Nothing                  -> Unknown `liftM` body
-            Just typ@P.UrlEncoded    -> sink UrlEncoded typ body
-            Just typ@(P.Multipart b) -> sink (Multipart b) typ body
-        c b s { actionReqBody = Just b }
+            config = actionConfig e
+            rbody = Wai.requestBody =<< requestSizeCheck (maxRequestSize config) req
+
+        b <- liftIO $ try(
+            case P.getRequestBodyType req of
+                Nothing                  -> sinkRaw rbody
+                Just typ@P.UrlEncoded    -> sinkUrlEncoded typ rbody
+                Just typ@(P.Multipart _) ->
+                    case uploadFilePath config of
+                        Nothing -> sinkMultipartLBS typ rbody
+                        Just p  -> sinkMultipartToDisk p typ rbody
+            )
+        case b of
+            Left (RequestSizeException limit) ->
```

Comment:
It's reasonable to add `Left _` with `status400` respond, but i don't think in any case there will be other kind exceptions since wai never throw in any case, so theoretically it's not necessary so i didn't add it.